### DISCUSSION TITLE
Capping the mintAmount per wallet address

### DISCUSSION THREAD

**Discussion by Jharmony** [hashlips-lab/nft-erc721-collection#249] [Ideas]
Is there a way to add a maximum mint amount per wallet address? For example, not allowing for individual wallets to be in more than 5 NFTs during the minting period. Maybe by adding another && operator and 
 `  modifier mintCompliance(uint256 _mintAmount) {
    require(_mintAmount > 0 && _mintAmount <= maxMintAmountPerTx **&&** _____ , 'Invalid mint amount!');
    require(totalSupply() + _mintAmount <= maxSupply, 'Max supply exceeded!');
    _;
  }`

Or maybe by creating another modifier function called maxMintAmount and adding the constraint there?

modifier maxMintAmount(uint256 _mintAmount) {
    require(msg.value >= maxMintAmount, 'You have reached the maximum mint amount per wallet.');
    _;
  }

Is it even possible? Many thanks for considering my request.

**Comment by liarco** [comment]
Hi @Jharmony this topic has been discussed many times, you can find a lot of info in #42. In that issue I explain the reason why I don't want to implement something like that in this project. I hope it helps.

Thank you for your understanding. 🙏

**Reply by Jharmony** [reply]
Thank you, @liarco. After reading your response to the issues posted on the link you provided, I agree that it is best to find a creative way of collecting the addresses to add them to the whitelist functionality to help save on the cost of deployment. I will proceed with this approach instead. Thank you for all of your help.
